acceptance/tests/pxp_module_bolt_download_file/download_file.rb

```diff
@@ -63,33 +68,74 @@ def test_file_does_not_exist(agent)
     on master, puppet('resource service puppetserver ensure=running')
   end
 
-  step 'execute successful download_file' do
-    clean_files(suts)
+  step 'execute successful download_file with files and directories' do
+    suts.each do |agent|
+      test_dir = test_dir_destination(agent)
+      test_symlink = test_file_destination(agent)
+      test_files = [
+        test_file_destination(agent),
+        test_file_destination(agent),
+        test_file_destination(agent)
+      ]
+      request = []
+      test_files.each do |file|
+        request << download_file_entry(@sha256, @source_file, '', file, 'file')
+      end
+      request << download_file_entry('', '', '', test_dir, 'directory')
+      request << download_file_entry('', '', test_files[0], test_symlink, 'symlink')
+      run_successful_download(master,
+                              agent,
+                              request)
+      test_files.each do |file|
+        test_file_resource_exists(agent, file, 'file')
+      end
+      test_file_resource_exists(agent, test_dir, 'directory')
+      test_file_resource_exists(agent, test_symlink, 'link')
+      teardown {
```

Comment:
does something else take care of cleaning non-file resources?

Reply:
Oh, whoops, no.